### ISSUE TITLE
[gfx1030][FP16][ROCm5.2] test_lrn_test failure due to hipRTC issue

### ISSUE THREAD

**Issue by junliume** [ROCm/MIOpen#1674]
**[How To Reproduce]:**
```
CXX=/opt/rocm/llvm/bin/clang++ CXXFLAGS='-Werror'  cmake -DMIOPEN_TEST_FLAGS=' --disable-verification-cache ' -DCMAKE_BUILD_TYPE=release -DBUILD_DEV=On -DMIOPEN_USE_MLIR=ON -DMIOPEN_GPU_SYNC=Off  -DMIOPEN_TEST_ALL=On -DMIOPEN_TEST_HALF=On -DCMAKE_PREFIX_PATH="/opt/rocm/;/opt/rocm/hip;/root/MIOpen/install_dir"  .. 

LLVM_PATH=/opt/rocm/llvm CTEST_PARALLEL_LEVEL=4 MIOPEN_CONV_PRECISE_ROCBLAS_TIMING=0 make -j 48 check
```

[Symptom]:
test_handle_test (Failed)
test_lrn_test (Failed)

**[More Details]:**
```
 4/108 Test  #32: test_lrn_test ..........................................***Failed  Error regular expression found in output. Regex=[FAILED]414.24 sec
/root/MIOpen/build/bin/test_lrn_test --half --input 1, 128, 56, 56 --N 1 --alpha 1 --beta 1 --K 1 --mode Across_Channel 
FAILED: /root/MIOpen/src/hipoc/hipoc_program.cpp:162: Failed creating module from file /tmp/miopen-interim-hsaco-083f-0ba5-deda-fab5/file shared object initialization failed
verify_lrn_bwd
Input Tensor Y 1, 128, 56, 56
Input Tensor DY 1, 128, 56, 56
Input Tensor X 1, 128, 56, 56
```
```
 29/108 Test  #27: test_handle_test .......................................***Failed    1.25 sec
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-b21968/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-ddd4a0/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.
```

**Comment by junliume** [comment]
@atamazov could you also take a look?

**Comment by junliume** [comment]
@muralinr from these two tickets that it might be the test's problem, could you check on it?
https://ontrack-internal.amd.com/browse/SWDEV-278594
https://ontrack-internal.amd.com/browse/SWDEV-290352
Could you check why this problem is coming back again?

**Comment by muralinr** [comment]
Hi @junliume I looked at these errors.  These "test_lrn_test" and "test_handle_test" failures are related to HIPRTC error compilation issues.  We should ask Artem or Paul to look at this issue.

test_lrn_test ..........................................***Failed  Error regular expression found in output. Regex=[FAILED]414.24 sec
/root/MIOpen/build/bin/test_lrn_test --half --input 1, 128, 56, 56 --N 1 --alpha 1 --beta 1 --K 1 --mode Across_Channel 
FAILED: /root/MIOpen/src/hipoc/hipoc_program.cpp:162: Failed creating module from file /tmp/miopen-interim-hsaco-083f-0ba5-deda-fab5/file shared object initialization failed

Test  #27: test_handle_test .......................................***Failed    1.25 sec
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-b21968/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-ddd4a0/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.

**Comment by shurale-nkn** [comment]
@junliume  @atamazov 
Always happens during testing in our CI. 
CI can't detect failures because of regexp in upper case

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/46b85e8f2dcaaee26b9cacbaf6a9babe2335f5b5/test/CMakeLists.txt#L333

```
"test_handle_test" start time: Aug 10 19:34 UTC
Output:
----------------------------------------------------------
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-5caf37/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.
Error: Failed to compile source (from CL or HIP source to LLVM IR).

MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-51ea2f/input/CompileSource:6:10: error: use of undeclared identifier 'num'
    data[num] *= 2;
         ^
1 error generated when compiling for gfx1030.
Error: Failed to compile source (from CL or HIP source to LLVM IR).

MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' nop_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: nop_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-72bce5/input/CompileSource:5:28: error: unused parameter 'data' [-Werror,-Wunused-parameter]
__global__ void write(int* data) {
                           ^
1 error generated when compiling for gfx1030.
Error: Failed to compile source (from CL or HIP source to LLVM IR).

MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-67a46a/input/tinygemm.cl:1:48: error: use of undeclared identifier 'i'
__kernel void write(__global int* data) { data[i] = 0; }
                                               ^
1 error generated.

MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-0fabf1/input/tinygemm.cl:1:48: error: use of undeclared identifier 'i'
__kernel void write(__global int* data) { data[i] = 0; }
                                               ^
1 error generated.

MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-86efe8/input/tinygemm.cl:1:35: error: unused parameter 'data' [-Werror,-Wunused-parameter]
__kernel void write(__global int* data) {}
                                  ^
1 error generated.

<end of output>
```

**Comment by junliume** [comment]
@muralinr it seems that we need to reopen one of the above mentioned tickets. It should be assigned to compiler I think.

**Comment by junliume** [comment]
> @junliume @atamazov Always happens during testing in our CI. CI can't detect failures because of regexp in upper case
> 
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/46b85e8f2dcaaee26b9cacbaf6a9babe2335f5b5/test/CMakeLists.txt#L333
> 
> ```
> "test_handle_test" start time: Aug 10 19:34 UTC
> Output:
> ----------------------------------------------------------
> MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
> MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
> MIOpen(HIP): Warning [BuildHip] /tmp/comgr-5caf37/input/CompileSource:6:10: error: use of undeclared identifier 'num'
>     data[num] *= 2;
>          ^
> 1 error generated when compiling for gfx1030.
> Error: Failed to compile source (from CL or HIP source to LLVM IR).
> 
> MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' error_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
> MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: error_hip.cpp
> MIOpen(HIP): Warning [BuildHip] /tmp/comgr-51ea2f/input/CompileSource:6:10: error: use of undeclared identifier 'num'
>     data[num] *= 2;
>          ^
> 1 error generated when compiling for gfx1030.
> Error: Failed to compile source (from CL or HIP source to LLVM IR).
> 
> MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' nop_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
> MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: nop_hip.cpp
> MIOpen(HIP): Warning [BuildHip] /tmp/comgr-72bce5/input/CompileSource:5:28: error: unused parameter 'data' [-Werror,-Wunused-parameter]
> __global__ void write(int* data) {
>                            ^
> 1 error generated when compiling for gfx1030.
> Error: Failed to compile source (from CL or HIP source to LLVM IR).
> 
> MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
> MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
> MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-67a46a/input/tinygemm.cl:1:48: error: use of undeclared identifier 'i'
> __kernel void write(__global int* data) { data[i] = 0; }
>                                                ^
> 1 error generated.
> 
> MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
> MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
> MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-0fabf1/input/tinygemm.cl:1:48: error: use of undeclared identifier 'i'
> __kernel void write(__global int* data) { data[i] = 0; }
>                                                ^
> 1 error generated.
> 
> MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
> MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
> MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-86efe8/input/tinygemm.cl:1:35: error: unused parameter 'data' [-Werror,-Wunused-parameter]
> __kernel void write(__global int* data) {}
>                                   ^
> 1 error generated.
> 
> <end of output>
> ```

Thanks @shurale-nkn  I think we need to fix this testing defect too! :)

**Comment by junliume** [comment]
Taken from official documentation, shall we change the line to the follwing?
```
set_tests_properties(${TEST_NAME} PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED;[^a-z]Error;ERROR;Failed")

**Comment by shurale-nkn** [comment]
> Taken from official documentation, shall we change the line to the follwing?
> 
> ```
> set_tests_properties(${TEST_NAME} PROPERTIES FAIL_REGULAR_EXPRESSION "FAILED;[^a-z]Error;ERROR;Failed")
> ```
@junliume 
no, that would be too easy. We are not looking for easy ways 😄 

The real problem is the variability of the words used in our output. `Error` can be used in Verification to define difference between gold value and result, it also can be a part of compilation argument. Therefore, we can't say that it is always a negative marker.

**Comment by shurale-nkn** [comment]
On the one hand, if the error is not critical and the program can perform a task by another method, it is not necessary to complete it. But we should be able to track it in our CI, where in that case we should always get Error message. 

So the mechanism is similar to the current behavior of the `assert` function. Which works only in debug builds. But this should work in any build if some flag was defined.

**Comment by atamazov** [comment]
[Informative] Current convention is that the failing test should print `FAIL` (uppercase) or return non-zero code.

**Comment by atamazov** [comment]
@junliume Build errors in `test_handle_test` is not a bug. Among other checks, this test **_intentionally causes build errors_** in order to check if the library properly catch build errors. Error messages on the console are normal.

**Comment by atamazov** [comment]
I am looking into `test_lrn_test` problem. Is it rocm or device specific?

**Comment by shurale-nkn** [comment]
> @junliume Build errors in `test_handle_test` is not a bug. Among other checks, this test **_intentionally causes build errors_** in order to check if the library properly catch build errors. Error messages on the console are normal.

In this case, the CI should check the output data for an error and check the correctness of the handling.

**Comment by atamazov** [comment]
### Update on `test_handle_test` failure
`error: use of undeclared identifier 'num'` has nothing to do with the failure. The test prints this onto console correctly; this is normal behavior (as I mentioned at https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1674#issuecomment-1214218691).

Actual error message if somehow hidden by ctest (when you run `make check` or `ctest --output-on-failure -C ./test -R test_handle_test`).

But if you run just `./bin/test_handle_test` you will see:
```bash
FAILED: throws([&] { h.AddKernel("NoAlgo", "", "nop_hip.cpp", "write", {1, 1, 1}, {1, 1, 1}, "", 0, false, WriteNop(miopenHIPKernelType)); }): /atamazov-github/MLOpen1/test/handle_test.cpp: 227
Aborted (core dumped)
```
I am looking into this.

**Comment by atamazov** [comment]
1. The reason of the issue is that starting from ROCm 5.2 the HIP compiler (both HIPRTC and COMgr paths) does not print this warning:
```bash
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' nop_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: nop_hip.cpp
MIOpen(HIP): Warning [BuildHip] /tmp/comgr-78cc19/input/CompileSource:5:28: error: unused parameter 'data' [-Werror,-Wunused-parameter]
```
for the source code like this
```C++
__global__ void write(int* data) {
}
```
The handle test properly detects the missing warning and fails.

Note that what we are testing in the "warning test" is capability of the compiler to issue warnings. We do not care if the compiler warns about specific situation in the code -- it is up to the compiler developers. Therefore while the reason of the issue is a compiler change, this is not a compiler bug. We can add more "dangerous" code to the testing kernel in order to enforce the compiler to issue warnings and reduce the probability of similar issues in the future.

2. The reason why this issue was not caught by QA is that they use release non-DEV builds, but this specific test case is skipped in non-DEV the _warning test_ it can't work (`-Wno-everything` is passed to HIP and OCL compilers in non-DEV builds to avoid problems for our end users). But DEV builds test this and we see CI failures.

3. The reason of mess during triage is that diagnostics needs to be improved.

I am going to fix the 1st and the 3rd problems. The 2nd problem cannot be resolved.

**Comment by atamazov** [comment]
The previous comment is incorrect.

@junliume In which docker container have you found the issue with test_handle_test?

**Comment by atamazov** [comment]
[Informative] I've used 5.2 container with MIOpen installed. Rebuilding the library didn't matter because tests still use the installed library. The tests were built with BUILD_DEV=On and _warning test_ was enabled. But the installed library was built with BUILD_DEV=Off and unable to issue build warnings. Therefore the test has failed, but the failures were actually false positives, and I got the impression that there are compiler problems in 5.2.

**Comment by junliume** [comment]
> The previous comment is partially incorrect.
> 
> @junliume In which docker container have you found the issue with test_handle_test?

@atamazov  I am using a standalone gfx1030 node with base OS (bare metal) ROCm-5.2.0

**Comment by atamazov** [comment]
@junliume Thank you. I have gfx1030 on hand, but the base driver is 4.3.0, so may I need to upgrade the node. Please update me with your results with a docker.

**Comment by junliume** [comment]
**[ENV]**:
base: ROCm-5.2.0
docker: latest mainline docker

**[Observations]**:

1. Category 1: CMake Error undefined symbol.  `test_immed_conv3d` `test_conv2` `test_conv3D`
    e.g.
    ```
    /root/MIOpen/build/bin/test_immed_conv3d: symbol lookup error: /root/MIOpen/build/bin/test_immed_conv3d: undefined symbol: _ZN6miopen20LoadOrPrepareInvokerERNS_6HandleERNS_18ConvolutionContextENS_6solver2IdENS_4conv9DirectionE, version MIOPEN_HIP_1
    CMake Error at test_test_immed_conv3d.cmake:7 (message):
      Test failed
    ```
3. Category 2: error: use of undeclared identifier 'num' `test_handle_test`
    e.g.

**Comment by junliume** [comment]
@atamazov it looks all related to `handle` issues which may or may not be related to hipRTC

**Comment by atamazov** [comment]
[Informative] Not reproducible with 5.2 release docker and 4.3.0 base driver.

**Comment by junliume** [comment]
@atamazov let's put this issue on hold then. I just checked my last run with this docker and cannot reproduce the issues either.

We are trying to update base OS ROCm to 5.2 and docekr ROCm to the same, let's see if gfx1030-fp16 stability has changed.

Thanks!

**Comment by atamazov** [comment]
## Results of running `test_handle_test` with Mainline build 5.3.0-10659

⚠️ The error shown at https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1674#issuecomment-1217183297 is most likely because of installed MIOpen library was built with BUILD_DEV=Off, while the test was built with BUILD_DEV=On, see https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1674#issuecomment-1215731028. **_The following results were obtained with MIOpen removed from /opt/rocm_**:

(A) When MIOpen is build with BUILD_DEV=Off and installed, the test passes.

(B) When MIOpen is build with BUILD_DEV=On, the test fails with this:
```bash
root@Drakkar:~/MIOpen/build# ctest --output-on-failure -C ./test -R test_handle_test
Test project /root/MIOpen/build
    Start 27: test_handle_test
1/1 Test #27: test_handle_test .................***Failed    0.59 sec
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1030
MIOpen(HIP): Info [Handle] stream: 0x16e8180, device_id: 0
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1030
MIOpen(HIP): Info [Handle] stream: 0x16e8180, device_id: 0
MIOpen(HIP): Info [PrintVersion] HIPRTC v.9.0
MIOpen(HIP): Error [Compile] 'hiprtcCompileProgram(prog.get(), c_options.size(), c_options.data())' test_hip.cpp: HIPRTC_ERROR_COMPILATION (6)
MIOpen(HIP): Error [BuildHip] HIPRTC status = HIPRTC_ERROR_COMPILATION (6), source file: test_hip.cpp
MIOpen(HIP): Warning [BuildHip] In file included from <built-in>:1:
/tmp/comgr-da295e/include/hiprtc_runtime.h:1:5: error: this style of line directive is a GNU extension [-Werror,-Wgnu-line-marker]
# 1 "/long_pathname_so_that_rpms_can_package_the_debug_info/src/out/ubuntu-20.04/20.04/build/hip-on-rocclr/src/hiprtc/hip_rtc_gen/hipRTC_header.h"
    ^
/long_pathname_so_that_rpms_can_package_the_debug_info/src/out/ubuntu-20.04/20.04/build/hip-on-rocclr/src/hiprtc/hip_rtc_gen/hipRTC_header.h:1:5: error: this style of line directive is a GNU extension [-Werror,-Wgnu-line-marker]
# 1 "<built-in>" 1
    ^
/long_pathname_so_that_rpms_can_package_the_debug_info/src/out/ubuntu-20.04/20.04/build/hip-on-rocclr/src/hiprtc/hip_rtc_gen/hipRTC_header.h:5:5: error: this style of line directive is a GNU extension [-Werror,-Wgnu-line-marker]
# 1 "/long_pathname_so_that_rpms_can_package_the_debug_info/src/external/hip-on-vdi/include/hip/hip_runtime.h" 1 3
    ^
/long_pathname_so_that_rpms_can_package_the_debug_info/src/out/ubuntu-20.04/20.04/build/hip-on-rocclr/src/hiprtc/hip_rtc_gen/hipRTC_header.h:6:5: error: this style of line directive is a GNU extension [-Werror,-Wgnu-line-marker]
# 1 "/long_pathname_so_that_rpms_can_package_the_debug_info/src/external/hip-on-vdi/include/hip/hip_fp16.h" 1 3
    ^
4 errors generated when compiling for gfx1030.
terminate called after throwing an instance of 'miopen::Exception'
  what():  /root/MIOpen/src/hipoc/hipoc_program.cpp:299: Code object build failed. Source: test_hip.cpp
CMake Error at test_test_handle_test.cmake:7 (message):
  Test failed
```

The root reason of the issue seems to using some GNU extension during preparations of precompiled headers + recent changes in the HIP compiler, namely https://reviews.llvm.org/D124534 (which is intended to fix https://github.com/llvm/llvm-project/issues/55067) which warns about usage GNU extension.

I am working on isolating the issue and providing a fix or workaround.

**Comment by atamazov** [comment]
@junliume Please add `non_miopen_bug`.

I see some issues in pooling kernels (build errors due to unused variables). I will continue fixing the issues until 5.2 DEV builds pass `make check`.

**Comment by atamazov** [comment]
@junliume In DEV builds, `test_lrn_test` and `test_activation` fail due to memory allocation error. Tested on both gfx906 (16GB) and Navi21 (16GB), on different ROCm versions (5.0.0 and Mainline build 10659) with different base drivers. Non-DEV builds pass! Reason is still unknown. I am working on the workaround.

```bash
   640  /root/MIOpen/build/bin/test_lrn_test --half --input 1, 16, 4096, 4096 --N 5 --alpha 1 --beta 1 --K 1 --mode Within_Channel
   641  FAILED: /root/MIOpen/src/hip/handlehip.cpp:85: Memory not available to allocate buffer: 536870912
...
   431  1: /root/MIOpen/build/bin/test_activation --half --input 1, 16, 4096, 4096 --alpha 0.95 --beta 2.3 --gamma 3.4 --mode CLIPPEDRELU --packed 0
   432  1: FAILED: /root/MIOpen/src/hip/handlehip.cpp:85: Memory not available to allocate buffer: 536870912
```

**Comment by shurale-nkn** [comment]
This is SWDEV-345683.
PR1729 is workaround for this issue, main problem in HIP. Fixed in 5.3.0-27.
necessary to arrange changes as workaround and continue throwing exceptions after hip 5.3.0-27 in order to monitor the unexpected behavior of the system.

**Comment by atamazov** [comment]
@shurale-nkn What is this SWDEV-345683 about? Some quirks of hipMemGetInfo or hipFree or...?

@junliume I can prepare the followup PR but I need 5.3.0-27 where the problem is expected to be fixed.

I recommend removing `help_wanted` and setting `urgency_low` or `normal` (as this affects tests only).

**Comment by ppanchad-amd** [comment]
@junliume Is this fixed with latest ROCm 6.0.2 (HIP 6.0.32831)? Thanks!

**Comment by atamazov** [comment]
@ppanchad-amd I think this may be closed.
